### PR TITLE
First draft of Component Structure section

### PR DESCRIPTION
**Pull request by davidkpiano**
@HugoGiraudel 

This is the first draft of the Component Structure section, from https://github.com/HugoGiraudel/sass-guidelines/issues/146 . Let me know what you think! :smile:

### CODE REVIEW

File: en/_architecture.md [hunk 1]
```diff
@@ -34,6 +34,22 @@ For instance, a search form should be treated as a component. It should be reusa
 
 Most of any interface can be thought of as little components and I highly recommend you stick to this paradigm. This will not only shorten the amount of CSS needed for the whole project, but also happens to be much easier to maintain than a chaotic mess where everything is flustered.
 
+## Component Structure
+
+Ideally, components should exist in their own Sass partial (within the `components/` folder, as is described in the 7-1 pattern), such as `components/_button.scss`. The styles described in each component file should only be concerned with:
```
Comment (HugoGiraudel) [inline en/_architecture.md]: We could add a link to *7-1 pattern*.

File: en/_architecture.md [hunk 2]
```diff
@@ -34,6 +34,22 @@ For instance, a search form should be treated as a component. It should be reusa
...
+
+* the style of the component itself;
+* the style of the component's variants, modifiers, and/or states;
```
Comment (HugoGiraudel) [inline en/_architecture.md]: Smart quote. ;)

File: en/_architecture.md [hunk 3]
```diff
@@ -34,6 +34,22 @@ For instance, a search form should be treated as a component. It should be reusa
...
+* the styles of the component's descendents (i.e. children), if necessary.
```
Comment (HugoGiraudel) [inline en/_architecture.md]: Smart quote. ;)

File: en/_architecture.md [hunk 4]
```diff
@@ -34,6 +34,22 @@ For instance, a search form should be treated as a component. It should be reusa
...
+
+If you want your components to be able to be themed externally (e.g. from a theme inside the `themes/` folder), limit the declarations to only structural styles, such as dimensions (width/height), padding, margins, alignment, etc. Exclude styles such as colors, shadows, font rules, background rules, etc.
+
+A component partial can include component-specific `$variables`, `%placeholders`, and even `@mixin`s and functions. Keep in mind, though, that you should avoid referencing (i.e. `@import`ing) component files from other component files, as this can make your project's dependency graph an unmaintainable tangled mess.
```
Comment (HugoGiraudel) [inline en/_architecture.md]: I don’t think we need code styles here. Plain words are fine enough. :)

Comment (HugoGiraudel) [inline en/_architecture.md]: Smart quote. ;)

**Comment by HugoGiraudel** [comment]
Thank you very much David, I really dig it. I’d merge it in 1.3.0 tomorrow and add props for it. :)

**Comment by HugoGiraudel** [comment]
Didn’t wait until tomorrow, haha. Would you mind giving me your opinion: https://github.com/HugoGiraudel/sass-guidelines/commit/82ce787344fe6616b19607f2e0d49541061babdf?

**Comment by davidkpiano** [comment]
Looks good to me! Thanks for the revisions! Should we remove the code style for "...`@import`-ing" too?

**Comment by HugoGiraudel** [comment]
I wanted to keep it.

**Comment by HugoGiraudel** [comment]
Closing. :)

**Pull request closed without merging**